### PR TITLE
Like and bookmark

### PR DESCRIPTION
**Pull request by shraddha-amber**
This PR contains:

Post like and unlike functionality book
Add and remove post from bookmark

### CODE REVIEW

**Review by Tarandeep-s1ngh — APPROVED** [review]
`Only one suggestion:` On the bookmarks page, you can display the bookmarked posts such that the latest one appears on the top.

Everything else looks great!

- Easy to understand
- Clean File structure
- Dividing the code into multiple components
- Liked how you have used barrel export

Good job!

**Comment by shraddha-amber** [comment]
> `Only one suggestion:` On the bookmarks page, you can display the bookmarked posts such that the latest one appears on the top.
> 
> Everything else looks great!
> 
> * Easy to understand
> * Clean File structure
> * Dividing the code into multiple components
> * Liked how you have used barrel export
> 
> Good job!

Thanks @Tarandeep-s1ngh implementing it now

File: src/hooks/useLikePost.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  const dispatch = useDispatch();
+  const [isLiked, setLiked] = useState(false);
+  const { user: authUser, token } = useSelector((store) => store.auth);
+  const { posts } = useSelector((store) => store.posts);
+
+  useEffect(() => {
+    (() => {
+      if (
+        Boolean(
+          post.likes.likedBy.find(
```
Comment (aniketxparihar) [inline src/hooks/useLikePost.js:15]: try to use ?. after objects in case they return empty they won't break the screen.

**Review by aniketxparihar — APPROVED** [review]
- Very Clean and Understandable Code
- I liked how you maintained the loading states in Redux
- naming conventions used are on point
- Code is well structured and Modular
Good Job!

File: src/hooks/useLikePost.js (hunk 1 above)
Comment (shraddha-amber) [reply src/hooks/useLikePost.js:15]: yes, I mostly use them where there's a possibility that the property won't be present

**Pull request merged**